main/externalData/balances/index.ts

```diff
@@ -178,34 +183,58 @@ export default function (store: Store) {
     if (changedBalances.length > 0) {
       store.setBalances(address, changedBalances)
 
-      const knownTokens = storeApi.getKnownTokens(address)
+      const knownTokens = new Set(storeApi.getKnownTokens(address).map(toTokenId))
 
       // add any non-zero balances to the list of known tokens
       const unknownBalances = changedBalances
-        .filter(b => parseInt(b.balance) > 0 && !knownTokens.some(t => t.address === b.address && t.chainId === b.chainId))
+        .filter(b => parseInt(b.balance) > 0 && !knownTokens.has(toTokenId(b)))
 
       if (unknownBalances.length > 0) {
         store.addKnownTokens(address, unknownBalances)
       }
 
       // remove zero balances from the list of known tokens
-      const zeroBalances = changedBalances
-        .filter(b => parseInt(b.balance) === 0 && knownTokens.some(t => t.address === b.address && t.chainId === b.chainId))
-
-      if (zeroBalances.length > 0) {
+      const zeroBalances = changedBalances.reduce((zeroBalSet, b) => {
```

Comment:
do we need reduce here? seems like we're just filtering. if you want to just change this into a set you can just wrap the previous call like this:

```suggestion
      const zeroBalances = new Set(
        changedBalances.filter(b => parseInt(b.balance) === 0 && knownTokens.has(tokenId))
      )
```